### PR TITLE
Disable kernel auto startup in untrusted workspace

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #6077

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1900,7 +1900,7 @@
         "build-ipywidgets-compile": "tsc -p ./src/ipywidgets && rimraf ./out/tsconfig.tsbuildinfo && node ./src/ipywidgets/scripts/copyfiles.js",
         "build-ipywidgets-webpack": "cross-env NODE_OPTIONS=--max_old_space_size=9096 webpack --config ./src/ipywidgets/webpack.config.js",
         "checkDependencies": "gulp checkDependencies",
-        "postinstall": "npm run download-api && node ./build/ci/postInstall.js",
```
Comment (DonJayamanne) [inline package.json:1903]: This should be done either as a separate job (optional that can fail).
Else we're all unable to submit PRs even when VS Code Insiders hasn't been updated.

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -192,6 +192,67 @@ jobs:
       - name: Check dependencies
         run: npm run checkDependencies
 
+  vsc_api_check:
```
Comment (DonJayamanne) [inline .github/workflows/build-test.yml:195]: Created a separate Job that will test compilation against the latest type definition files.
This way we can have CI running and perfectly green until VS Code actually ships stuff.

Comment (DonJayamanne) [reply .github/workflows/build-test.yml:195]: For instance, I should have be able to run CI right now with the current VSC Insiders.

Comment (DonJayamanne) [reply .github/workflows/build-test.yml:195]: This will allow us to have both

File: package.json (hunk 1 above)
Comment (DonJayamanne) [reply package.json:1903]: @joyceerhl @IanMatthewHuff /cc

**Review by IanMatthewHuff — APPROVED** [review]

**Pull request merged**